docs/cryptodoc/src/05_08_ml_dsa.rst

```diff
@@ -0,0 +1,443 @@
+   +------------+-----------+-----------+---------------+-----------+--------------+-----------------+------------------+------------------+---------------+--------------+---------------+----------------+
+   | ML-DSA-6x5 | 8380417   | 256       | 1753          | 13        | 49           | 192             | 2\ :sup:`19`     | :math:`(q-1)/32` | (6,5)         | 4            | 196           | 55             |
+   +------------+-----------+-----------+---------------+-----------+--------------+-----------------+------------------+------------------+---------------+--------------+---------------+----------------+
+   | ML-DSA-8x7 | 8380417   | 256       | 1753          | 13        | 60           | 256             | 2\ :sup:`19`     | :math:`(q-1)/32` | (8,7)         | 2            | 120           | 75             |
+   +------------+-----------+-----------+---------------+-----------+--------------+-----------------+------------------+------------------+---------------+--------------+---------------+----------------+
+
+The parameter sets claim a NIST security level of 2, 3, and 5 respectively.
+
+.. _pubkey/ml_dsa/internals:
+
+Algorithm Internals
+-------------------
+
+[FIPS-204]_ describes three primary operations: key generation, signing, and
+verification.
+
+Internally, those operations are further split into two functional layers:
+the public-facing ML-DSA (Section 5), and ML-DSA-internal (Section 6). ML-DSA
+and ML-DSA-internal decouple the actual high-level logic from the generation of
+randomness. The functions in ML-DSA-internal receive pre-determined random bytes
+as needed and are, therefore, fully deterministic.
+
+ML-DSA is a Schnorr-like signature where the typically interactive protocol is
+made non-interactive by pseudorandomly deriving the verifier's challenge from
+the commitment and the message to be signed.
+
+.. _pubkey/ml_dsa/components:
+
+.. table:: ML-DSA Components and File Locations
+
+   +-------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------+--------------------+
+   | Component                                                                                                                                             | Purpose                                                        | [FIPS-204]_        |
+   +=======================================================================================================================================================+================================================================+====================+
+   | :ref:`Types <pubkey/ml_dsa/types>` (:srcref:`src <[src/lib/pubkey/dilithium/dilithium_common]/dilithium_types.h>`)                                    | Strong Types                                                   | n/a                |
```

Comment:
Personally, I like this variant to link to GitHub. Maybe we can discuss in our next JF if we want to adapt this to all overview lists.

Reply:
Yeah, simply using `(src)` really is a crutch here to safe space in the rendering. `.../dilithium_symmetric_primitives.h` really is too wide for this table. ;) I like the idea of using a symbol here, perhaps even integrate it as a feature in `:srcref:`. Maybe `:srcref-icon:`, or perhaps as a special case when the entire URL is marked as ellipsis using brackets, like: ``:srcref:'[src/lib/pubkey/dilithium/dilithium_common/dilithium_types.h]'``

Reply:
Please open a ticket for this.